### PR TITLE
tests/e2e/k8s: Add more CRD-mode tests

### PR DESCRIPTION
**Pull request by orozery**
This PRadds further e2e tests testing the CRD-based controlplane.
It mostly tests the control component of the controlplane.

### CODE REVIEW

File: tests/e2e/k8s/test_import.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+	impService := &util.Service{
+		Name: imp.Name,
+		Port: imp.Spec.Port,
+	}
+
+	_, err = cl[0].AccessService(httpecho.GetEchoValue, impService, true, &services.ConnectionResetError{})
+	require.ErrorIs(s.T(), err, &services.ConnectionResetError{})
+
+	require.Nil(s.T(), cl[0].DeleteService(service.Name))
```
Comment (kfirtoledo) [inline tests/e2e/k8s/test_import.go:131]: Can you add comment what you do in each part

File: tests/e2e/k8s/util/kind.go [hunk 2]
```diff
@@ -488,6 +506,61 @@ func (c *KindCluster) ScaleDeployment(name, namespace string, replicas int32) er
...
+		}
+
+		cond := meta.FindStatusCondition(*statusConditions, conditionType)
+		if cond == nil {
+			return false, nil
+		}
+
+		switch cond.Status {
+		case metav1.ConditionFalse:
+			return !conditionStatus, nil
```
Comment (kfirtoledo) [inline tests/e2e/k8s/util/kind.go:539]: Maybe it is better to change it to `expectCondition' ==false

File: tests/e2e/k8s/test_import.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+		Namespace: namespace,
+		Port:      imp.Spec.Port,
+	}
+
+	_, err = cl[0].AccessService(httpecho.GetEchoValue, impService, true, &services.ConnectionResetError{})
+	require.ErrorIs(s.T(), err, &services.ConnectionResetError{})
+
+	require.Nil(s.T(), cl[0].Cluster().Resources().Delete(context.Background(), imp))
+	require.Nil(s.T(), cl[0].Cluster().WaitForDeletion(service))
+	require.Nil(s.T(), cl[0].Cluster().WaitForDeletion(systemService))
```
Comment (kfirtoledo) [inline tests/e2e/k8s/test_import.go:226]: you didn't delete it in line 202? maybe move it to there

File: tests/e2e/k8s/test_import.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+	require.Nil(s.T(), cl[0].Cluster().Resources().Get(context.Background(), service.Name, service.Namespace, service))
+	service.Labels[control.LabelManagedBy] = "other"
+	require.Nil(s.T(), cl[0].Cluster().Resources().Update(context.Background(), service))
+	require.Nil(s.T(), cl[0].WaitForImportCondition(imp, v1alpha1.ImportServiceCreated, false))
+
+	service.Labels[control.LabelManagedBy] = control.AppName
+	require.Nil(s.T(), cl[0].Cluster().Resources().Update(context.Background(), service))
+	require.Nil(s.T(), cl[0].WaitForImportCondition(imp, v1alpha1.ImportServiceCreated, true))
+	_, err = cl[0].AccessService(httpecho.GetEchoValue, impService, true, &services.ConnectionResetError{})
```
Comment (kfirtoledo) [inline tests/e2e/k8s/test_import.go:144]: What is the purpose of this test it failed and you just change the labels

**Review by kfirtoledo — CHANGES_REQUESTED** [review]

**Comment by orozery** [comment]
@kfirtoledo PTAL

**Review by kfirtoledo — APPROVED** [review]

**Pull request merged**